src/GitVersionCore/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -61,8 +61,12 @@ void UpdatePreReleaseTag(GitVersionContext context, SemanticVersion semanticVers
             if (tagToUse == "useBranchName")
             {
                 Logger.WriteInfo("Using branch name to calculate version tag");
-                var name = branchNameOverride ?? context.CurrentBranch.Name;
-                tagToUse = name.RegexReplace(context.Configuration.BranchPrefixToTrim, string.Empty, RegexOptions.IgnoreCase);
+                tagToUse = branchNameOverride ?? context.CurrentBranch.Name;
+                if (!string.IsNullOrWhiteSpace(context.Configuration.BranchPrefixToTrim))
```

Comment:
Why the change to this logic since this shouldn't be related to your issue (only the RegexReplace afterwards should be enough to make it work). Or does this fix another issue?